compaction/compaction_manager.cc

```diff
@@ -584,7 +584,12 @@ void compaction_manager::submit(column_family* cf) {
             column_family& cf = *task->compacting_cf;
             sstables::compaction_strategy cs = cf.get_compaction_strategy();
             sstables::compaction_descriptor descriptor = cs.get_sstables_for_compaction(cf, get_candidates(cf));
-            int weight = calculate_weight(descriptor.sstables);
+            // Use weight 0 for compactions that are comprised solely of completely expired sstables.
+            // We want these compactions to be in a separate weight class because they are very lightweight, fast and efficient.
+            int weight = 0;
+            if (descriptor.is_compact_fully_expired == sstables::compaction_descriptor::compact_fully_expired::no) {
+                weight = calculate_weight(descriptor.sstables);
+            }
```

Comment:
can be rewritten as:

`int weight = descriptor.compact_fully_expired ? 0 : calculate_weight(descriptor.sstables);`

perhaps we should now pass descriptor to calculate_weight, so calculate_weight itself will return 0 if it sees compact_fully_expired set to true. so this knowledge can be reused elsewhere if needed.

Reply:
+1